Updating setup instructions and adding data guides

schunkes opened the pull request:
This PR includes an update to the setup instructions, including a quick start guide which goes through the setup procedure without explaining it in detail. Additionally I added a short section to the `data-guide` section, which explains how to add manually downloaded data sets to Eradiate and a table that can serve as an index to those files.

# Description

*Describe your feature. What does it do? How does it work? If applicable mention an issue on the issue tracker*

# Checklist

- [x] The code follows the relevant coding guidelines
- [x] The code generates no new warnings
- [x] The code is apropriately documented.
- [x] The code is tested to prove its function
- [x] The feature branch is rebased on the current state of the `main` branch
- [x] I give permission that the Eradiate project may redistribute my contributions under the terms of its license

nollety commented on docs/rst/user_guide/data_guide.rst:
```diff
@@ -18,6 +18,34 @@ library, whose data model is based on NetCDF. xarray provides a comprehensive,
+     - Description
+     - Download link
+   * - ``us76-u86-4-fullrange``
+     - US76 standard atmosphere below 86km with four molecules
```
I would write:

> Absorption cross sections for a gas mixture of N2, O2, CO2 and CH4 where the mixing ratio correspond to the US76 standard atmosphere model in the altitude region below 86 km.

to indicate what molecules are there and what variable is in the data set, but I assume that's a bit too much text for the table?

schunkes replied:
Yes, I think that's too long for this table. But I agree, we should have a more detailed index and documentation of the datasets we ship. Did you talk to Vincent about something like that before?

nollety replied:
Perhaps something similar to [how solar irradiance datasets are described](https://eradiate.readthedocs.io/en/latest/rst/user_guide/solar_irradiance_spectra.html)?
No, I did not talk to Vincent about that. I feel this is not so urgent and I try not to disturb him during this busy period ^^

nollety commented on docs/rst/getting_started/building.rst:
```diff
@@ -283,44 +322,11 @@ the embedded Mitsuba version number to the terminal:
-
-   We are not sourcing the script, we are executing it in a subshell.
+Eradiate does not automatically ship all available data sets due to their size.
+In order to successfully run all tests and tutorials, at least the ``us76_u86_4-fullrange``
```
The ``us76_u86_4-fullrange`` data set is not required to run tests, right?
I'm curious, what tutorials require this dataset? I think the OneDimSolver tutorial requires it but that's the only one, no?

schunkes replied:
We talked about that already so I'll just close this

nollety replied:
If the xarray tutorial in the user guide requires that the `us76_u86_4` datasets are downloaded, maybe we should add to that tutorial a reference to the section `_sec-user_guide-manual_download` in the data guide?

nollety reviewed (commented):
Hi @schunkes
I reviewed the parts about the data download only. I think it's nice, I just have some questions 😃

schunkes commented:
Alright, I updated the setup guide and added a quickstart guide to the docs as well.

Additionally, I created a page for molecular absorption datasets. This page holds a table of all available data sets with a short description and a link to our web server, hosting the file. On top of that the page holds a section with detailed information on every dataset.

Lastly, I added a section to the `data guide` chapter, explaining how to install the datasets downloaded from our website.

leroyvn commented:
Merged! Thanks for the update!
